prometheus/histogram.go

```diff
@@ -309,41 +311,30 @@ func (h *histogram) Write(out *dto.Metric) error {
 	// This is a bit arcane, which is why the following spells out this if
 	// clause in English:
 	//
-	// If the currently-hot counts struct is #0, we atomically increment
-	// h.countAndHotIdx by 1 so that from now on Observe will use the counts
-	// struct #1. Furthermore, the atomic increment gives us the new value,
-	// which, in its most significant 63 bits, tells us the count of
+	// Adding 1<<63 switches the hot index (from 0 to 1 or from 1 to 0)
+	// without touching the count bits. After the atomic operation finishes
+	// Observe will use the counts struct #1.
+	// Furthermore, the atomic increment gives us the new value,
+	// which, in its least significant 63 bits, tells us the count of
 	// observations done so far up to and including currently ongoing
 	// observations still using the counts struct just changed from hot to
 	// cold. To have a normal uint64 for the count, we bitshift by 1 and
-	// save the result in count. We also set h.hotIdx to 1 for the next
-	// Write call, and we will refer to counts #1 as hotCounts and to counts
-	// #0 as coldCounts.
+	// save the result in count.
 	//
 	// If the currently-hot counts struct is #1, we do the corresponding
 	// things the other way round. We have to _decrement_ h.countAndHotIdx
 	// (which is a bit arcane in itself, as we have to express -1 with an
 	// unsigned int...).
```

Comment:
This whole comment block can be simplified even more. It still contains parts that only are true for the old way and not the nice new way you found. As things are now less arcane, perhaps it's also better to have a line-by-line comment for explanation.

Reply:
What do you think of my description at <https://github.com/pascaldekloe/metrics/blob/master/metrics.go#L144-L156>? Copy/paste over here?

Reply:
Looks good. I'd probably add more explanations to the lines below, but I can do that later, too.